src/Native/System.Native/pal_process.h

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include <stdint.h>
+
+/**
+ * Used by System.Diagnostics.Process.Start to fork/exec a new process.
+ *
+ * This function takes the place of directly using fork and execve from managed code,
+ * in order to avoid executing managed code in the child process in the window between
+ * fork and execve, which is not safe.
+ *
+ * As would have been the case with fork/execve, a return value of 0 is success and -1
+ * is failure; if failure, error information is provided in errno.
+ */
+extern "C"
+int32_t ForkAndExecProcess(
+    const char* filename,    // filename argument to execve
+    char* const argv[],      // argv argument to execve
+    char* const envp[],      // envp argument to execve
+    const char* cwd,         // path passed to chdir in child process
+    int32_t redirectStdin,    // whether to redirect standard input from the parent
+    int32_t redirectStdout,   // whether to redirect standard output to the parent
+    int32_t redirectStderr,   // whether to redirect standard error to the parent
+    int32_t* childPid,        // [out] the child process' id
+    int32_t* stdinFd,         // [out] if redirectStdin, the parent's fd for the child's stdin
+    int32_t* stdoutFd,        // [out] if redirectStdout, the parent's fd for the child's stdout
+    int32_t* stderrFd);       // [out] if redirectStderr, the parent's fd for the child's stderr
+
+
+/**
+ * These values differ from OS to OS, so make a constant contract.
+ * These values apply for the current process only
+ */
+enum class RLimitResources : int32_t
+{
+    PAL_RLIMIT_CPU          = 0,        // CPU limit in seconds
+    PAL_RLIMIT_FSIZE        = 1,        // Largest file that can be created, in bytes
+    PAL_RLIMIT_DATA         = 2,        // Maximum size of data segment, in bytes
+    PAL_RLIMIT_STACK        = 3,        // Maximum size of stack segment, in bytes
+    PAL_RLIMIT_CORE         = 4,        // Largest core file that can be created, in bytes
+    PAL_RLIMIT_AS           = 5,        // Address space limit
+    PAL_RLIMIT_RSS          = 6,        // Largest resident set size, in bytes
+    PAL_RLIMIT_MEMLOCK      = 7,        // Locked-in-memory address space
+    PAL_RLIMIT_NPROC        = 8,        // Number of processes
+    PAL_RLIMIT_NOFILE       = 9,        // Number of open files
+};
+
+/**
+ * The current and maximum resource values for the current process.
+ * These values are depict the resource according to the above enum.
+ */
+struct RLimit
+{
+    uint64_t            CurrentLimit;
```

Comment:
Nit: style choice,  but I wouldn't create so much white space to line up things that aren't even in the same struct/function.